api/src/main/java/ai/djl/util/ZipUtils.java

```diff
@@ -121,14 +119,15 @@ private static void addToZip(Path root, Path file, ZipOutputStream zos) throws I
         }
     }
 
-    static String removeLeadingFileSeparator(String name) {
-        int index = 0;
-        for (; index < name.length(); index++) {
-            if (name.charAt(index) != File.separatorChar) {
-                break;
-            }
+    static void validateArchiveEntry(String name, Path destination) throws IOException {
+        Path expectedOutputPath = destination.resolve(name).normalize();
+        if (!expectedOutputPath.startsWith(destination.normalize())) {
```

Comment:
I think we'd better block ".." as well in this method. it prevent file overwrite inside the destination folder. In original version, we already blocking "..", and nobody complained about it.

Reply:
The default behavior of extracting a tar is to overwrite isn't it? I'm not sure why we need to differ.

If an archive had (in order)
```
b.txt
a/../b.txt
```

a/../b.txt would overwrite b.txt

Reply:
I've gone ahead and added it back to keep in line with what we had before, but curious to know what you think about my point above.